src/suffer.cpp

```diff
@@ -954,65 +1058,41 @@ void suffer::from_sunburn( Character &you )
-        //~ number of body parts
-        message = n_gettext( "The sunlight burns your %s.",
-                             "The sunlight burns your %s.",
-                             affected_bodyparts.size() );
-    }
-    you.add_msg_if_player( m_bad, message, all_parts_list );
+    // s is a list of body parts.  The plurality integer is the total
+    // number of body parts plus a correction if the eyes are included
+    const int plurality = affected_part_names.size() + contains_eyes;
 
-    // Wake up from skin irritation/burning
-    if( you.has_effect( effect_sleep ) ) {
-        you.wake_up();
-    }
+    auto warn_and_wake_up = [ &you, &all_parts_list, &plurality]
+    ( std::string singular, std::string plural, game_message_type type ) {
+        std::string message = n_gettext( singular.c_str(), plural.c_str(), plurality );
 
-    // Solar Sensitivity (SUNBURN) trait causes injury to exposed parts
-    if( you.has_trait( trait_SUNBURN ) ) {
-        you.mod_pain( 1 );
-        // Check exposure of all body parts
-        for( const std::pair<const bodypart_id, float> &bp_exp : bp_exposure ) {
-            const bodypart_id &this_part = bp_exp.first;
-            const float exposure = bp_exp.second;
-            // Skip parts with adequate protection
-            if( exposure <= MIN_EXPOSURE ) {
-                continue;
-            }
-            // Don't damage eyes directly, since it takes from head HP (in other words, your head
-            // won't be destroyed if only your eyes are exposed).
-            if( this_part == bodypart_id( "eyes" ) ) {
-                continue;
-            }
-            // Exposure percentage determines likelihood of injury
-            // 10% exposure is 10% chance of injury, naked = 100% chance
-            if( x_in_y( exposure, 1.0 ) ) {
-                // Because hands and feet share an HP pool with arms and legs, and the mouth shares
-                // an HP pool with the head, those parts take an unfair share of damage in relation
-                // to the torso, which only has one part.  Increase torso damage to balance this.
-                if( this_part == bodypart_id( "torso" ) ) {
-                    you.apply_damage( nullptr, this_part, 2 );
-                } else {
-                    you.apply_damage( nullptr, this_part, 1 );
-                }
-            }
-        }
-    } else {
-        // Albinism/datura causes pain (1/60) or focus loss (59/60)
-        if( one_turn_in( 1_minutes ) ) {
-            you.mod_pain( 1 );
-        } else {
-            you.mod_focus( -1 );
+        you.add_msg_if_player( type, message, all_parts_list );
+        // Wake up from skin irritation/burning
+        if( you.has_effect( effect_sleep ) ) {
+            you.wake_up();
         }
+    };
+
+    switch( worst_effect ) {
+        case Damage:
+            warn_and_wake_up( "Your %s is bathed in sunlight.  It feels like it is burning up.",
```

Comment:
```suggestion
            warn_and_wake_up( n_gettext( "Your %s is bathed in sunlight.  It feels like it is burning up.", "the other text", the_number) );
```
The `n_gettext` marks the string literals for translation. Without it, they will never get translated.

Reply:
But `n_gettext` is already used within `warn_and_wake_up`. Should I move it out anyway for clarity?